### ISSUE TITLE
Autoformat packer with Stylua

### ISSUE THREAD

**Issue by akinsho** [wbthomason/packer.nvim#445]
@wbthomason thought I'd raise an issue to discuss before doing anything. In most of my other lua projects, I've recently added autoformatting on push/commits to the main branch using [stylua](https://github.com/JohnnyMorganz/StyLua). Not sure if you've encountered it, but it's a quite actively maintained, opinionated lua code formatter. I've found that it works quite well producing quite readable code as well as being great in CI since it has a github code action, so it's quite straightforward to setup, and means you'd generally no longer really need to think about how code that is contributed is formatted.

I know you've historically used `luaformatter` which I looked at initially when looking at options, but I've found it to be a bit inactive, harder to setup in CI and probably most subjectively produces quite hard to read code I find what it does to function arguments really hard to follow personally.

Not sure if it means much, but it's being adopted by quite a lot of other plugins and projects in the nvim space, including I think a preliminary draft PR in neovim itself to maybe adopt it as well.

In either case I have set it up on several of my own projects so should be pretty straightforward to do here.

As a side note, I was initially hesitant to adopt it because it seemed strict about the `use {value}` syntax trying to encompass it in parens, but the author added in an option which allows single parameter functions to go without braces.

Anyway, let me know what you think, if you're open to it, I can raise a PR.

**Comment by wbthomason** [comment]
Thanks! I actually was unaware of `stylua`; `lua-format` has served me well so far, so I hadn't looked for alternatives.

I've tried out `stylua` a bit now. I definitely miss some of the more concise formatting preferences of `lua-format`, and don't wholly agree with the particular "opinionated" take `stylua` offers.

That said, I'm far from the only developer on this project anymore, and it would be useful to have a formatter integrated in CI. I'm ok switching over to `stylua` for this benefit, even if I don't love its style.

I added a tentative `stylua` config file in cab2f56. If you'd be willing to raise a PR to get it integrated into CI, that would be much appreciated.

**Comment by akinsho** [comment]
@wbthomason there's definitely choices it makes that I would do differently but since I personally hate to manually format my own code 😅 I spent a bit of time trying various options.

I think `stylua` whilst being opinionated in a way I don't always like is the most actively maintained (I hate the wasted space around the tables 😆), is the least offensive option IMO (subjective as that is).

The author has also been fairly responsive to the nvim community as he added the no call parentheses option after our badgering. Tbh I don't think he'd be open to adding loads of options for us, nor should he IMO. I think the value here is that despite no one being a 100% pleased with the output it will be fine for most people, similar to most other nvim projects (think telescope uses it now) and most importantly automatic.

**Comment by clason** [comment]
Small note: `stylua` is also being used for neovim itself: https://github.com/neovim/neovim/pull/14987 (nvim-lspconfig already switched, and so has nvim-treesitter).

A plugin can do its own thing, of course, but all else being equal, consistency with core neovim is a plus in my eyes.
